### PR TITLE
show verify and mdsnippets usage

### PR DESCRIPTION
**Pull request by SimonCropp**
I got some value out of this project. so i figured i would push back in a proposal

This PR uses a combination https://github.com/VerifyTests/Verify and https://github.com/SimonCropp/MarkdownSnippets to generate the result of a snippet, and then merge both that result and the snippet into the readme.md

Note that Verify  could also be used to simplify the assertions in the ConverterTests

I also took the liberty of doing some cleanup on your test csproj.

What do you think?

### CODE REVIEW

**Comment by SimonCropp** [comment]
if this is something you want to move forward with, let me know and i will fix the build

**Comment by mysticmind** [comment]
@SimonCropp Firstly, appreciate your efforts to give back to this project. Both Verify and MarkdownSnippets look to be quite an interesting thing. I am welcoming this with open arms 👍

File: README.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+<!--
+GENERATED FILE - DO NOT EDIT
```
Comment (SimonCropp) [inline README.md:2]: i can remove the header with config if you prefer

File: src/ReverseMarkdown/ReverseMarkdown.csproj [hunk 2]
```diff
@@ -4,15 +4,12 @@
     <VersionPrefix>1.0.0</VersionPrefix>
     <Authors>Babu Annamalai</Authors>
     <TargetFrameworks>net46;netstandard1.6;netstandard2.0</TargetFrameworks>
-    <AssemblyName>ReverseMarkdown</AssemblyName>
```
Comment (SimonCropp) [inline src/ReverseMarkdown/ReverseMarkdown.csproj:7]: these values default to the project name. so are redundant

**Comment by SimonCropp** [comment]
build fixed. 

do u want a github action https://github.com/SimonCropp/MarkdownSnippets/blob/master/docs/github-action.md

File: README.md (hunk 1 above)
Comment (mysticmind) [reply README.md:2]: Yes, I would prefer not to have that header in the README file. I will give it a shot to add the config.

File: src/ReverseMarkdown/ReverseMarkdown.csproj (hunk 2 above)
Comment (mysticmind) [reply src/ReverseMarkdown/ReverseMarkdown.csproj:7]: Agree!

**Comment by mysticmind** [comment]
> 
> 
> build fixed.
> 
> do u want a github action https://github.com/SimonCropp/MarkdownSnippets/blob/master/docs/github-action.md

with regards to the github actions, will defer it for now since there are not many changes which go into readme.

**Comment by mysticmind** [comment]
@SimonCropp I am observing few issues on the generated markdown which I would like to highlight to you:
- Generated anchor tag has some issues, i.e. `<a id='snippet-usage'/></a>`,  this anchor tag has a `/` (which should not be there) in the opening `a` tag followed by an end tag. Also the `id` value should be idiomatically double quotes 
- In the generated code snippet i.e. `string html = "This a sample <strong>paragraph</strong> from <a href=\"http://test.com\">my site</a>";`, the href value has `\` to escape the quotes which should not be there.

Could you please check and revert?

**Comment by SimonCropp** [comment]
> In the generated code snippet i.e. string html = "This a sample <strong>paragraph</strong> from <a href=\"http://test.com\">my site</a>";, the href value has \ to escape the quotes which should not be there.

why should it not be there? it is not valid c# if those are removed

**Comment by SimonCropp** [comment]
> Also the id value should be idiomatically double quotes

I would prefer not to. they are semantically equivalent. and  the c# code used to generating html using single quotes is more readable, since u dont need to constantly escape. and given this file should not be edited the single quotes should not be seen

**Comment by SimonCropp** [comment]
> this anchor tag has a / (which should not be there) in the opening a

i have pushed a fix for this

**Comment by mysticmind** [comment]
> > In the generated code snippet i.e. string html = "This a sample **paragraph** from <a href="[http://test.com\">my](http://test.com%5C%22%3Emy) site";, the href value has \ to escape the quotes which should not be there.
> 
> why should it not be there? it is not valid c# if those are removed
You are right, I mistook it to be html and not C#. My bad, ignore my comment.

**Comment by mysticmind** [comment]
> > Also the id value should be idiomatically double quotes
> 
> I would prefer not to. they are semantically equivalent. and the c# code used to generating html using single quotes is more readable, since u dont need to constantly escape. and given this file should not be edited the single quotes should not be seen

Okay, understood.

**Comment by SimonCropp** [comment]
Did u want a PR that converts your tests to using snapshots?

**Comment by mysticmind** [comment]
> 
> 
> Did u want a PR that converts your tests to using snapshots?

Can you please elaborate on this so that I get a better understanding of the usage and the benefits?

Also with regards to the markdown snippets, for cases like README files, it would help to have a single file rather than a source and generated file. I would like a feature which works out of a single markdown rather than 2 files which will ease of managing it. Essentially, it boils down to either update the snippet if you find `snippet: Usage` or `<!-- snippet: Usage -->` and `<!-- endsnippet -->` pairs on the same markdown file. Thoughts?

**Comment by mysticmind** [comment]
> 
> 
> > Also the id value should be idiomatically double quotes
> 
> I would prefer not to. they are semantically equivalent. and the c# code used to generating html using single quotes is more readable, since u dont need to constantly escape. and given this file should not be edited the single quotes should not be seen

I tend to agree with your thoughts since the html is generated on c# side and using single quote makes the code easier to read on the c# side. In retrospect, I should messaged it as "I would prefer using idiomatic double quotes".

BTW, I did read through the twitter thread on this.

**Pull request merged**